python_modules/libraries/dagster-components/dagster_components_tests/unit_tests/test_component_rendering.py

```diff
@@ -0,0 +1,68 @@
+from typing import Optional, Sequence
+
+import pytest
+from dagster_components.core.component_rendering import (
+    TemplatedValueResolver,
+    _should_render,
+    preprocess_value,
+)
+from pydantic import BaseModel, Field, TypeAdapter
+
+
+class Inner(BaseModel):
+    a: Optional[str] = None
+    deferred: Optional[str] = Field(default=None, json_schema_extra={"defer_render": True})
```

Comment:
So this `defer_render` thing will be part of the public api?

Reply:
that's correct yeah -- although we can add some sugar on top of it to make it a little more user-friendly, i.e.:

```python
from dagster_components import defer_render

class Inner(BaseModel):
    deferred: Optional[str] = defer_render(Field(default=None))

```